packages/website/ts/components/portal/portal.tsx

```diff
@@ -448,13 +465,17 @@ export class Portal extends React.Component<PortalProps, PortalState> {
         const newScreenWidth = utils.getScreenWidth();
         this.props.dispatcher.updateScreenWidth(newScreenWidth);
     }
+    private _isSmallScreen(): boolean {
+        const result = this.props.screenWidth === ScreenWidths.Sm;
+        return result;
```

Comment:
If this type of stuff is what you were thinking about making a Higher Order Component then I 100% agree (not now though)

Reply:
I was thinking specifically about generalizing / abstracting away the mechanism of alerting the dispatcher of screen width updates which is baked into the portal component now: https://github.com/0xProject/0x-monorepo/blob/a26cc7c14d4e59102d5f84d534f75c42b8683cad/packages/website/ts/components/portal/portal.tsx#L293

Reply:
Yea I _think_ we're on the same page? Basically it should be possible to dependency inject stuff like screen width and these helper methods into any arbitrary component. We can do this using redux and connect if we want (dispatching the new screen width in some root component or something), or just bypassing that and just do the listening in the HOC. The argument for not bypassing would be that we can claim that App = fn(local state, redux store) and is therefore more predictable / pure / etc...